lib/ansible/modules/cloud/azure/azure_rm_localnetworkgateway.py

```diff
@@ -0,0 +1,246 @@
+        required: true
+    name:
+        description:
+            - Name of local network Gateway.
+        required: true
+    state:
+        description:
+            - Assert the state of the local Gateway. Use 'present' to create or update local network gateway and
+              'absent' to delete local network gateway.
+        default: present
+        choices:
+            - absent
+            - present
+        required: false
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+        required: false
+    local_address_prefixes:
+        description:
+            - List of local network site address space(on-premise).
+    gateway_ip_address:
+        description:
+            - IP address of local network gateway.
+    bgp_settings:
+        description:
+            - Local network gateway's BGP speaker settings.
+        suboptions:
+            asn:
+                description:
+                    - The BGP speaker's ASN.
+            bgp_peering_address:
+                description:
+                    - The BGP peering address and BGP identifier of this BGP speaker.
+            peer_weight:
+                description:
+                    - The weight added to routes learned from this BGP speaker.
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Madhura Naniwadekar (@Madhura-CSI)"
+'''
+
+EXAMPLES = '''
+    - name: Create local network gateway, with BGP settings
+      azure_rm_localnetworkgateway:
+        resource_group: testrg
+        name: testlocalgw
+        local_address_prefixes:
+          - 10.1.0.0/16
+          - 172.100.0.0/16
+        gateway_ip_address: 1.2.3.4
+        tags:
+          common: a
+      register: local_network_gateway
+
+    - name: Delete local network gateway
+      azure_rm_localnetworkgateway:
+        resource_group: testrg
+        name: testlocalgw
+        state: absent
+'''
+
+RETURN = '''
+id:
+    description: The ID of local network gateway.
+    returned: always
+    type: str
+    sample: /subscriptions/xxxx/resourceGroups/xxx/providers/Microsoft.Network/localNetworkGateways/xx
+'''
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase, CIDR_PATTERN
+
+
+bgp_spec = dict(
+    asn=dict(type='int'),
+    bgp_peering_address=dict(),
+    peer_weight=dict(type='int', default=0)
+)
+
+
+class AzureRMLocalNetworkGateway(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            location=dict(type='str'),
+            local_address_prefixes=dict(type='list', default=[]),
```

Comment:
what if user want to remove all existing local_address_prefixes setting?

Reply:
User simply needs to remove local_address_prefixes part from module call. This will clear the configuration from local network gateway and module state will be 'changed = true'